geo/Geoserver.py

```diff
@@ -396,7 +396,7 @@ def create_workspace(self, workspace: str):
     def upsert_workspaces_rules(self, workspacePattern: str = '*', permission: str = 'r', role: str = None):
         """
         Create a new security rule for either all workspaces or a subset of them based on the provided pattern,
-        or update an existing security rule if it already exists.
+        or update an existing security rule if it already exists. One role can only be set in one rule.
```

Comment:
actually this is not true, the current restriction is that every rule only accepts one role, but one role could be in more than one rule

Reply:
Fixed!